# Phylogenetic-Tree-Building-Algorithms
Phylogenetic trees offer researchers a clear understanding of the evolution of
species over timeâ€”numerous algorithms for creating trees with varying degrees of accuracy and
completeness. Phylogenetic trees are an essential part of many biological studies, and the choice
of algorithm can significantly impact the accuracy and interpretation of Results. The main issue
with the algorithms is that there is a wide range of tree-building algorithms available, each with its
strengths and weaknesses. Researchers often face challenges in selecting the most appropriate
method for their specific research questions. Guiding algorithm selection can help researchers to
make informed decisions, ultimately leading to more accurate and meaningful results. We
carefully selected and designed input sequences from the MLH1 gene across plant and animal
species to answer specific biological questions for Convergent evolution, divergent evolution,
homologous sequences, and distantly related sequences. We implemented the UPGMA
(Unweighted Pair Group Method with Arithmetic Mean) and (Neighbor Joining) NJ algorithms in
Python. We used Bio Python to perform Multiple sequence alignment on the input sequences and
then built the appropriate distance matrix. We compared the performance of the studied algorithms
for accuracy and computation time. Our results demonstrate the strengths and limitations of each
algorithm while answering the specific biological question such as convergent, divergent, homologous and non-homologous evolution.
